packages/blockchain-wallet-v4-frontend/src/data/modules/sfox/sagas.js

```diff
@@ -281,33 +286,57 @@ export default ({ coreSagas }) => {
     }
   }
 
-  const getJumioStatus = function*() {
+  const completeJumio = function*() {
     try {
-      yield put(A.getJumioStatusLoading())
+      // Jumio status of 'PENDING' can mean the user finished verification
+      // flow and the results are 'PENDING' jumio. Or the user has not finished
+      // and the flow requires user action.
+      // When a Jumio callback has been retreived, set jumio token to 'complete'
+      // so the UI can determine which type of 'PENDING' status the user is.
+      const tokenR = yield select(selectors.core.kvStore.buySell.getSfoxJumio)
+      const accountsR = yield select(selectors.core.data.sfox.getAccounts)
+      const accounts = accountsR.getOrElse([])
+      let token = tokenR.getOrElse({})
```

Comment:
would `getOrFail('Could not retrieve token')` be better here? I think that token is needed for the ongoing process. Perhaps a refetch/restart would be required

Reply:
Maybe I should just call completeJumio with the token in the payload? The user needs to have a token, otherwise they would not have been able to complete the jumio flow.

Reply:
Sounds very reasonable
There should ideally be some flow if token is not present

Reply:
Use getOrFail as tony suggests